pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -599,6 +599,18 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 				return reconcile.Result{Requeue: true}, nil
 			}
 		}
+		// Set the Provisioned status condition for adopted clusters (and in case we upgraded to/past where that condition was introduced)
```

Comment:
Shouldn't this be in the above `if` block where we are setting the installedtimestamp to ensure this only happens for already installed (and/or adopted) clusters?

Reply:
No; installed clusters will have their installedTimestamp set already.

Reply:
I meant setting the provisioned condition to true only if cd is already installed

Reply:
I'm missing something.

The new code block is inside the `if cd.Spec.Installed {` block started on L593. So it will only be triggered when `cd.Spec.Installed` is True.

It is _outside_ of the block where we check whether `InstalledTimstamp` is set because we want to trigger both setting that timestamp and setting the Provisioned condition based on `cd.Spec.Installed` being True.

Reply:
D'oh - github formatting mistake - sorry!